### PR TITLE
Swift: Build incompatible OS diagnostic on all platforms.

### PR DESCRIPTION
**Pull request by sashabu**

### CODE REVIEW

**Review by geoffw0 — COMMENTED** [review]
How is this / can this be tested?

File: swift/BUILD.bazel [hunk 1]
```diff
@@ -77,12 +77,12 @@ pkg_filegroup(
...
     ] + select({
-        "@platforms//os:linux": [
-            ":incompatible-os",
-        ],
         "@platforms//os:macos": [
             ":xcode-autobuilder",
         ],
+        "//conditions:default": [
+            ":incompatible-os",
+        ],
```
Comment (redsun82) [inline swift/BUILD.bazel:85]: shouldn't this be something like
```bazel
pkg_filegroup(
    name = "extractor-pack-arch",
    srcs = select({
        "@platforms//os:windows": [],
        "//conditions:default": [
            ":extractor",
            ":swift-test-sdk-arch",
        ],
    }) + select({
        "@platforms//os:macos": [
            ":xcode-autobuilder",
        ],
        "//conditions:default": [
            ":incompatible-os",
        ],
    }),
    visibility = ["//visibility:public"],
)
```
or should this be postponed to a follow-up PR where the whole `extractor-pack` can be built on Windows (including a Windows version of `extractor-pack-generic`, probably with just an `autobuild.cmd`?

Comment (sashabu) [reply swift/BUILD.bazel:85]: Yes, I was planning on doing this in a follow-up PR, but see my comment to Geoff below: I'm now leaning towards adding all the commits here instead.

**Comment by sashabu** [comment]
> How is this / can this be tested?

I tested locally on Windows with `bazel build //swift/tools/autobuilder-diagnostics:incompatible-os`. Before this PR, it fails; with this PR it succeeds.

This is probably not useful on its own, but will be required for emitting nice error messages on Windows. The error messages will have an integration test when implemented.

If you (or anyone else) are not comfortable merging this without automated testing, I'm happy to hold off and put everything in the same PR. That might be a good idea in light of Paolo's comment as well - perhaps in my aim to have small, self-contained PRs, I've gone too far and sent out *half* of a PR!

**Comment by sashabu** [comment]
> > If you (or anyone else) are not comfortable merging this without automated testing
> 
> No, I'm happy at this stage to understand how this has been and can be tested. It will be good to have automated coverage later.

I'll close this in favour of https://github.com/github/codeql/pull/13447 which puts this in the context of the rest of the extractor pack (which has an internal PR with tests).

**Pull request closed without merging**